src/Bicep.Local.Extension/Host/Attributes/TypeAnnotationAttribute.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Azure.Bicep.Types.Concrete;
+
+namespace Bicep.Local.Extension.Host.Attributes;
+[AttributeUsage(AttributeTargets.Property)]
+public class TypeAnnotationAttribute : Attribute
+{
+    public TypeAnnotationAttribute(
```

Comment:
[nit] What about `TypePropertyAttribute` to align with naming for object properties?